### PR TITLE
Add PositionUpdater component doc, and update columns references docs

### PR DESCRIPTION
**Pull request by jolelievre**
Add PositionUpdater component doc, and update columns references docs
Add documentation for:
- `PrestaShop\PrestaShop\Core\Grid\Position\GridPositionUpdater`
and its general use

- `PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\ToggleColumn`
- `PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\ImageColumn`
- `PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\BadgeColumn`
- `PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\LinkColumn`
- `PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\PositionColumn`

and update other columns to match the layout (use table to display options)

### CODE REVIEW

**Review by mickaelandrieu — APPROVED** [review]

File: src/content/1.7/development/components/grid/columns-reference/link.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+| **route_param_name** | string | **required** Parameter name used by the route to generate the url. |
+| **route_param_field** | string | **required** Record field containing the route parameter. |
+
+## Example usage
+
+```php
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\LinkColumn;
+use PrestaShop\PrestaShop\Core\Grid\Column\ColumnCollection;
+
+$linkColumn = new LinkColumn('total_spent');
```
Comment (PierreRambaud) [inline src/content/1.7/development/components/grid/columns-reference/link.md]: Change the identifier and the name to be consistent with others?

File: src/content/1.7/development/components/grid/columns-reference/employee-name-with-avatar.md [hunk 2]
```diff
@@ -25,13 +30,12 @@ Employee's name field.
 use PrestaShop\PrestaShop\Core\Grid\Column\Type\Employee\EmployeeNameWithAvatarColumn;
 use PrestaShop\PrestaShop\Core\Grid\Column\ColumnCollection;
 
-$column = new EmployeeNameWithAvatarColumn('severity');
-$column->setName('Severity (1-4)');
-$column->setOptions([
-     'field' => 'severity',
-     'with_message' => true,    // enable severity messages
+$employeeColumn = new EmployeeNameWithAvatarColumn('employee');
```
Comment (PierreRambaud) [inline src/content/1.7/development/components/grid/columns-reference/employee-name-with-avatar.md]: Did not see this class `EmployeeNameWithAvatarColumn` pass in a PR :/
Avatar should have been an option :/

File: src/content/1.7/development/components/position-updater/_index.md [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+    arguments:
+      - 'product'
+      - 'id_product'
+      - 'position'
+      - 'id_category_default'
+```
+
+## Building your PositionUpdate
+
+The good news is that you now made the hardest part, all other computing and request will now be managed by our component.
```
Comment (PierreRambaud) [inline src/content/1.7/development/components/position-updater/_index.md]: `requests`?

File: src/content/1.7/development/components/grid/columns-reference/employee-name-with-avatar.md (hunk 2 above)
Comment (mickaelandrieu) [reply src/content/1.7/development/components/grid/columns-reference/employee-name-with-avatar.md]: +1, who have validated this new column type? this break MVC principles /C @sarjon @matks

File: src/content/1.7/development/components/grid/columns-reference/link.md (hunk 1 above)
Comment (jolelievre) [reply src/content/1.7/development/components/grid/columns-reference/link.md]: yep my bad it was copied from badge

File: src/content/1.7/development/components/grid/columns-reference/employee-name-with-avatar.md (hunk 2 above)
Comment (jolelievre) [reply src/content/1.7/development/components/grid/columns-reference/employee-name-with-avatar.md]: you're probably right, but then the code should be fixed not the doc ^^

File: src/content/1.7/development/components/position-updater/_index.md (hunk 3 above)
Comment (jolelievre) [reply src/content/1.7/development/components/position-updater/_index.md]: queries better, thanks

**Comment by mickaelandrieu** [comment]
Thanks @jolelievre !

**Pull request merged**